### PR TITLE
Add write response status code tests for DataStorage write operation

### PR DESCRIPTION
**Pull request by mirahsupreme**
Verifies that the correct WriteResponse and WriteResponseStatus is returned for both successful and failed write operations.

### CODE REVIEW

File: test/TestDataStorageWrite.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import factorapi.DataStorage;
+import factorapi.WriteResponse;
+import factorapi.WriteRequest;
+import org.mockito.Mockito;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (Oliver-222) [inline test/TestDataStorageWrite.java]: Using the &apos;.*&apos; form of import should be avoided - checkstyle

**Review by Oliver-222 — COMMENTED** [review]
/home/runner/work/Software-Engineering-Project/Software-Engineering-Project/test/TestDataStorageWrite.java:17: error: cannot find symbol
        Mockito.when(successResponse.getStatus()).thenReturn(WriteResponse.WriteResponseStatus.SUCCESS);
                                    ^
  symbol:   method getStatus()
  location: variable successResponse of type WriteResponse
/home/runner/work/Software-Engineering-Project/Software-Engineering-Project/test/TestDataStorageWrite.java:23: error: cannot find symbol

> Task :compileTestJava FAILED
4 actionable tasks: 4 executed
            assertEquals(WriteResponse.WriteResponseStatus.SUCCESS, response.getStatus(), "Write operation should succeed");
                                                                            ^
  symbol:   method getStatus()
  location: variable response of type WriteResponse
/home/runner/work/Software-Engineering-Project/Software-Engineering-Project/test/TestDataStorageWrite.java:[30](https://github.com/Oliver-222/Software-Engineering-Project/actions/runs/12236303475/job/34129455061?pr=58#step:5:31): error: cannot find symbol
        Mockito.when(failureResponse.getStatus()).thenReturn(WriteResponse.WriteResponseStatus.FAILURE);
                                    ^
  symbol:   method getStatus()
  location: variable failureResponse of type WriteResponse
/home/runner/work/Software-Engineering-Project/Software-Engineering-Project/test/TestDataStorageWrite.java:[35](https://github.com/Oliver-222/Software-Engineering-Project/actions/runs/12236303475/job/34129455061?pr=58#step:5:36): error: cannot find symbol
            assertEquals(WriteResponse.WriteResponseStatus.FAILURE, response.getStatus(), "Write operation should fail");
                                                                            ^
  symbol:   method getStatus()
  location: variable response of type WriteResponse
4 errors

File: test/TestDataStorageWrite.java (hunk 1 above)
Comment (loganullmeyer) [inline test/TestDataStorageWrite.java]: This is what is triggering the checkstyle error, I don't think it likes when you use the ".*" for imports, should probably change to specific ones

**Comment by Oliver-222** [comment]
that was the error given by gradle

File: test/TestDataStorageWrite.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+public class TestDataStorageWrite {
+
+    @Test
+    public void testWriteResponseStatus() {
+        DataStorage ds = Mockito.mock(DataStorage.class);
+        
+        // Simulate WriteResponse with success status
+        WriteResponse successResponse = Mockito.mock(WriteResponse.class);
+        Mockito.when(successResponse.getStatus()).thenReturn(WriteResponse.WriteResponseStatus.SUCCESS);
```
Comment (loganullmeyer) [inline test/TestDataStorageWrite.java]: Apparently I forgot to actually add a getStatus() method to WriteResponse. You can just go in and create one, and this should compile as far as I can tell

**Review by loganullmeyer — COMMENTED** [review]
Two easy changes that I commented, otherwise LGTM

File: test/TestDataStorageWrite.java (hunk 2 above)
Comment (mirahsupreme) [reply test/TestDataStorageWrite.java]: i added it !

**Review by loganullmeyer — APPROVED** [review]
Wait to merge until getStatus PR is merged, otherwise LGTM

**Pull request merged**